### PR TITLE
Added basic child removal on remove_node

### PR DESCRIPTION
**Pull request by miversen33**
Note, I can't get the unit tests to run, this may not work as intended. Additionally, waiting for response on what to do with the child nodes on removal, right now they are tossed into the ether, that may not be ideal. Still, I think this accomplishes what is being asked in #227

### CODE REVIEW

**Comment by MunifTanjim** [comment]
Can you try running `./scripts/test.sh` locally? It should probably work (if you're using linux or mac).

**Comment by miversen33** [comment]
That works :) I will work through the error tossed quick and ping you when it passes my new test :)

**Comment by MunifTanjim** [comment]
btw, you can run `./scripts/test.sh nui/tree/init` to run test for that single file.

File: lua/nui/tree/init.lua [hunk 1]
```diff
@@ -344,6 +344,11 @@ end
 ---@return NuiTreeNode
 function Tree:remove_node(node_id)
   local node = self.nodes.by_id[node_id]
+  for _, child_id in ipairs(node._child_ids) do
+    -- We might want to store the nodes and return them with the node itself?
+    -- We should _really_ not be doing this recursively, but it will work for now
+    Tree:remove_node(child_id)
```
Comment (MunifTanjim) [inline lua/nui/tree/init.lua:350]: If the task of removing node from `self.nodes.by_id` is moved to a separate local function, we can avoid doing `vim.tbl_filter` on `node._child_ids` for each `child_id`. Something like:

```lua
local function remove_node(tree, node_id)
  -- remove node with the children recursively from tree.nodes.by_id` 
end
```

And then use it as:

```lua
function Tree:remove_node(node_id)
  local node = remove_node(self, node_id)
  -- rest of the function
end
```

Comment (miversen33) [reply lua/nui/tree/init.lua]: That makes sense. I migrated the node removal logic to its own local function and updated the PR to match

**Review by MunifTanjim — APPROVED** [review]
Thanks for noticing and fixing the bug. 😃

I'll be merging this shortly (don't worry about the lint errors, I'll take care of it).

**Comment by miversen33** [comment]
> Thanks for noticing and fixing the bug. smiley
> 
> I'll be merging this shortly (don't worry about the lint errors, I'll take care of it).

I already did ;) Also I pushed one more quick change to remove the recursion as it doesn't need to be there. Anyway! No problem, thanks for holding my hand lol

File: lua/nui/tree/init.lua [hunk 2]
```diff
@@ -340,11 +340,22 @@ function Tree:add_node(node, parent_id)
   self:_add_nodes({ node }, parent_node)
 end
 
+local function remove_node(tree, node_id)
+  local node = tree.nodes.by_id[node_id]
+  if node:has_children() then
+    for _, child_id in ipairs(node._child_ids) do
+      -- We might want to store the nodes and return them with the node itself?
+      tree.nodes.by_id[child_id] = nil
```
Comment (MunifTanjim) [inline lua/nui/tree/init.lua:348]: Umm... the previous code was right. Child nodes can also have children. For example, in `neo-tree` folders can have sub-folders of arbitrary depth.

Comment (MunifTanjim) [reply lua/nui/tree/init.lua:348]: > Also I pushed one more quick change to remove the recursion as it doesn't need to be there.

I think we do need the recursion.

Comment (miversen33) [reply lua/nui/tree/init.lua]: Oh ya, I kinda forgot about that. Whoops!

**Pull request merged**